arroyo-connectors/src/s3.rs

```diff
@@ -0,0 +1,129 @@
+use anyhow::{anyhow, bail, Result};
+use axum::response::sse::Event;
+use std::convert::Infallible;
+use typify::import_types;
+
+use arroyo_rpc::types::{ConnectionSchema, ConnectionType, TestSourceMessage};
+use arroyo_rpc::OperatorConfig;
+use serde::{Deserialize, Serialize};
+
+use crate::{pull_opt, Connection, EmptyConfig};
+
+use super::Connector;
+
+const TABLE_SCHEMA: &str = include_str!("../../connector-schemas/s3/table.json");
+
+import_types!(schema = "../connector-schemas/s3/table.json");
+
+pub struct S3Connector {}
+
+impl Connector for S3Connector {
```

Comment:
We're trying to be somewhat agnostic about which blob store is used for storing files. We use the object_store crate for this, and it'd be great if this new source could do the same. Right now blob store sinks fall under the `filesystem` name. A common pattern for connectors that have both source and sink implementations is to take a `type` argument which can be either `source` or `sink`, and then each one has its own set of options.

Reply:
Got it, will take a look 👍🏼